### PR TITLE
Point Proximity, Shape Proximity and Shape Cast in Havok

### PR DESCRIPTION
**Pull request by carolhmj**
This PR wraps these three Havok functionalities into methods on the plugin.

Point Proximity takes a IPointProximityQuery and stores its results in a ContactPoint representing the closest point on a body.
Shape Proximity takes a IShapeProximityQuery and stores its results in two ContactPoints, representing the hit point in shape space and world space.
Shape Cast takes a IShapeCastQuery and stores its results in two ContactPoints, representing the hit point in shape space and world space.

I'd love feedback here if the API and naming looks all right 🙂 One thing I'm not sure of is the `hitDistance` name in the ContactPoint class. For point prox and shape prox, it does really represent the distance to the hit point, but in the case of the shape cast, it is a fraction along the cast ray. So the name doesn't really describe it, but I'm not sure if it warrants a new class just to have a different name, or if there's a better name that works with both notions. I initially wanted to reuse the PhysicsRaycastResult class, but that also didn't work because that class has `hitPoint*World*` and one of the hit points is not in world space for shape prox/shape cast.

Here's some Playground examples:
Point Proximity: #I6AR8X#21
Shape Proximity: #1VT1BK#11
Shape Cast: #1VT1BK#12

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14736/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14736/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14736/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14736/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14736/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14736/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14736/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14736/merge#BCU1XR#0

**Comment by CedricGuillemet** [comment]
`ContactPoint` is very similar to `PhysicsRaycastResult`.
I would make it more generic this way:
Create a base class `CastingResult` with shared fields between ContactPoint and PhysicsRaycastResult. PhysicsRaycastResult would inherits from CastingResult. ContactPoint would be replaced by ProximityCastResult and ShapeCastResult. So you can make the differenciation between both regarding distance/fraction.

File: packages/dev/core/src/Physics/physicsShapeProximityQuery.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import type { Quaternion, Vector3 } from "../Maths/math.vector";
+import type { PhysicsShape } from "./v2/physicsShape";
+import type { PhysicsBody } from "./v2/physicsBody";
+
+/**
+ * Query for shape proximity.
+ */
+export interface IPhysicsShapeProximityQuery {
```
Comment (CedricGuillemet) [inline packages/dev/core/src/Physics/physicsShapeProximityQuery.ts:8]: I would name it IPhysicsProximityCastQuery instead, to make the difference with IPhysicsShapeCastQuery more obvious

**Review by RaananW — COMMENTED** [review]
Want to add the new wasm in this PR as wll? I was about to open a different PR, but if this one is already here maybe it will make sense to update it as part of this one.

**Review by CedricGuillemet — APPROVED** [review]
There is an error with the 1 st PG (Cannot read properties of undefined (reading 'transformNode')) . Other than that, LGTM!

**Comment by carolhmj** [comment]
> There is an error with the 1 st PG (Cannot read properties of undefined (reading 'transformNode')) . Other than that, LGTM!

@CedricGuillemet Oh, this is happening because the Havok plugin wasn't updated yet, it should go away after that!

**Pull request merged**